Sources/SwiftSyntax/SyntaxFactory.swift.gyb

```diff
@@ -82,7 +86,7 @@ public enum SyntaxFactory {
   }
 %   end
 
-%   if not node.is_base():
+%   if not node.is_base() and not node.must_uphold_invariant:
```

Comment:
Technically, I don’t think it’s correct to remove the `makeBlank` method if a node must uphold an invariant. 

In the future we might add invariants to other nodes as well that could have a valid blank content. One example that jumps to my mind would be string literals. For consistency, I would rather have this method return an Optional and return `nil` if the node kind has no valid blank representation.

@akyrtzi What do you think?

Reply:
Just to step back a bit here, what is the use case for the `makeBlank` methods, why do we need them? When would I call a `makeBlank` method instead of explicitly creating the syntax node that I want?

Reply:
Paging @harlanhaskins for my question.

Reply:
Ping @harlanhaskins

Reply:
Oh sorry! I think `makeBlank` was meant to be used for e.g. error recovery where we'd just fill in missing tokens with blank values in order to send a well-formed-but-invalid AST to semantic analysis.